src/contentmanager.cpp

```diff
@@ -617,26 +618,30 @@ QStringList ContentManager::getDownloadIds()
     return list;
 }
 
-void ContentManager::setCurrentLanguage(QString language)
+void ContentManager::setCurrentLanguage(QStringList languageList)
 {
-    if (language.length() == 2) {
-      try {
-        language = QString::fromStdString(
-                     kiwix::converta2toa3(language.toStdString()));
-      } catch (std::out_of_range&) {}
+    for (auto &language : languageList) {
+        if (language.length() == 2) {
+          try {
+            language = QString::fromStdString(
+                         kiwix::converta2toa3(language.toStdString()));
+          } catch (std::out_of_range&) {}
+        }
     }
-    if (m_currentLanguage == language)
-        return;
-    m_currentLanguage = language;
+    if (languageList.empty())
+        languageList.append("*");
+    m_currentLanguage = languageList.join(",");
+    KiwixApp::instance()->getSettingsManager()->setLanguage(languageList);
     emit(currentLangChanged());
     emit(filterParamsChanged());
 }
 
-void ContentManager::setCurrentCategoryFilter(QString category)
+void ContentManager::setCurrentCategoryFilter(QStringList categoryList)
 {
-    if (m_categoryFilter == category)
+    if (m_categoryFilter == categoryList.join(","))
```

Comment:
We should be prepared for list containing the same categories but in different orders.

Reply:
list is sorted before setting values now. So if there are same values, the order should be same and it equality check will work